engines/gob/sound/sound.cpp

```diff
@@ -337,7 +335,7 @@ void Sound::adlibPlayTrack(const char *trackname) {
 }
 
 void Sound::adlibPlayBgMusic() {
-	if (!_hasAdLib || _hasAdLibBg)
+	if (!_hasAdLib || !_hasAdLibBg) // If one of those is disabled, then stop there 
```

Comment:
If I understand your changes correctly, this is the only change needed to fix the bug you reported.
Could you minimize your commit to this change only?

Reply:
Sure, I updated the changes.